info/diktat-kotlin-coding-style-guide-en.md

```diff
@@ -941,46 +949,57 @@ This international code style prohibits non-latin (non ASCII) symbols in the cod
 
 #### <a name="r3.6"></a>Rule 3.6 line break style rules if the line is split
 
-  1. Compared to Java Kotlin allows not to to put semicolumn (';') after each statement separated by newline.
-     There should be no redundant semicolumn at the end of lines.
-     
-     In case when newline is needed to split the line, it should be placed after operators like &&/||/+/e.t.c
-     But newline should be placed before operators like ('.', '?.', '?:', '::', e.t.c), operator !! should not be separated from the value it is checking.
-     
-     Note, that you need to follow functional style:
-     ```kotlin
-        val value = otherValue!!
-                .map(x -> x)
-                .filter(true)
-                .size      
-     ```
-      
-  2. Newline should be placed after assignment operator ('=')
-  3. The name of a function or constructor should not be split by a newline from the opening brace '('. Brace follows immediately after the name without any spaces.   
-  4. Newline should be placed right after the comma (',')
-  5. If lambda expression has one-argument, a newline can appear after an arrow, if an expression has more than one argument before an arrow - breakpoint should appear after the brace.
+1. Compared to Java Kotlin allows not to to put semicolumn (';') after each statement separated by newline.
+ There should be no redundant semicolumn at the end of lines.
+ 
+ In case when newline is needed to split the line, it should be placed after operators like &&/||/+/e.t.c
+ But newline should be placed before operators like ('.', '?.', '?:', '::', e.t.c), operator !! should not be separated from the value it is checking.
```

Comment:
Also, why isn't `!!` listed with other operators that need newline after them? I don't see any differences in requirements. I guess separating `!!` won't even compile.

Reply:
```
a!! // here you use operator - it is separated from dot, it has absolutely different logic and meaning than `.?` or `.`
    .map{}
```

Reply:
`!!` differs completely with unbreakable operators like `?.` as it is not related to `.`

Reply:
Yes, I mean what is negative example in this case? What is 'separated from the value it is checking'? Is it something like
```
a
    !!.map { }
```
?

Reply:
yes, exactly 
a

Reply:
As far as I can tell, kotlin doesn't allow to write like this: `!!` can be separated from corresponding object by spaces, not by newlines. This example compiles:
```
a    !!.map { }
```
While your example doesn't.
Or am I missing something here?

Reply:
yes, looks that it cannot be compiled, but such code should be fixed to my mind:
```
a!!.map {}.map{}.map{}
```

to this one:
```
a!!
    .map{}
    .map{}
    .map{}
```

Is this the case for you?

Reply:
Yes, sure, it's just that wording was a little misleading at first.

Reply:
@petertrr please also be careful that you should not do the following:
```
a!!.map{}
    .map{}
    .map{}
```